app/widgets/common/deployModal/EnvironmentDropdown/EnvironmentDropdownList.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useMemo } from 'react';
+import { isEmpty } from 'lodash';
+import { Form } from 'cloudify-ui-components';
+import { Icon } from 'semantic-ui-react';
+import type { Environment } from './EnvironmentDropdown.types';
+import StageUtils from '../../../../utils/stageUtils';
+
+const translate = StageUtils.getT(
+    'widgets.common.deployments.deployModal.inputs.deploymentIdToDeployOn.environmentList'
+);
+
+const DATA_STATE = {
+    LOADING: 'LOADING',
+    EMPTY_LIST: 'EMPTY_LIST',
+    ITEM_LIST: 'ITEM_LIST'
+} as const;
+
+export interface EnvironmentDropdownListProps {
+    environments: Environment[];
+    onItemClick: (environment: Environment) => void;
+    activeEnvironmentId?: string;
+    isSuggestedList?: boolean;
+    loading?: boolean;
+}
+
+const EnvironmentDropdownList = ({
```

Comment:
I thought originally that we create here a list of dropdowns, but it seems like we are creating a list of dropdown items, so WDYT about:
```suggestion
const EnvironmentDropdownItemList = ({
```
?